vm/actor/src/builtin/power/types.rs

```diff
@@ -0,0 +1,410 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::DealWeight;
+use address::Address;
+use clock::ChainEpoch;
+use encoding::Cbor;
+use num_bigint::bigint_ser::{BigIntDe, BigIntSer};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use vm::{SectorSize, Serialized, TokenAmount};
+
+pub type SectorTermination = i64;
+
+/// Implicit termination after all deals expire
+pub const SECTOR_TERMINATION_EXPIRED: SectorTermination = 0;
+/// Unscheduled explicit termination by the miner
+pub const SECTOR_TERMINATION_MANUAL: SectorTermination = 1;
+
+#[derive(Clone)]
+pub struct SectorStorageWeightDesc {
```

Comment:
It might be worth adding comments to each of these types, I'll leave that to your judgment.

Reply:
I just don't know enough about them yet to add comments on all of them, I will make a pass through later when more specifics are ironed out